packages/audio-block/src/AudioBlock.tsx

```diff
@@ -0,0 +1,206 @@
+            setBlockSettings({
+                title: value,
+            });
+        }
+    };
+
+    const saveDescription = (value: string) => {
+        if (value !== blockSettings.description) {
+            setBlockSettings({
+                description: value,
+            });
+        }
+    };
+
+    const onRemoveAsset = () => {
+        deleteAssetIdsFromKey(AUDIO_ID, [audio?.id]);
+    };
+
+    const updateAudioAsset = (audio: Asset) => {
+        updateAssetIdsFromKey(AUDIO_ID, [audio.id]).then(() => setIsLoading(false));
+    };
+
+    const openAssetChooser = () => {
+        appBridge.openAssetChooser(
+            async (result) => {
+                setIsLoading(true);
+                updateAudioAsset(result[0]);
+                appBridge.closeAssetChooser();
+            },
+            {
+                multiSelection: false,
+                objectTypes: [AssetChooserObjectType.File],
+                extensions: AUDIO_EXTENSIONS,
+            }
+        );
+    };
+
+    useEffect(() => {
+        if (selectedFiles) {
+            setIsLoading(true);
+            uploadFile(selectedFiles[0]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedFiles]);
+
+    useEffect(() => {
+        if (droppedFiles) {
+            if (AUDIO_EXTENSIONS.includes(droppedFiles[0].name.split('.').pop() as FileExtension)) {
+                setIsLoading(true);
+                uploadFile(droppedFiles[0]);
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [droppedFiles]);
+
+    useEffect(() => {
+        if (doneAll && uploadResults) {
+            updateAudioAsset(uploadResults[0]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [doneAll, uploadResults]);
+
+    return (
+        <div data-test-id="audio-block" className={audioBlockClassNames}>
+            {audio ? (
+                <div className="tw-group">
+                    {isEditing && (
+                        <ItemToolbar
+                            onRemoveAsset={onRemoveAsset}
+                            onUploadClick={openFileDialog}
+                            onAssetChooseClick={openAssetChooser}
+                        />
+                    )}
+
+                    {isLoading ? (
+                        <div className="tw-flex tw-items-center tw-justify-center tw-h-14">
+                            <LoadingCircle />
+                        </div>
+                    ) : (
+                        <audio
+                            data-test-id="audio-block-audio-tag"
+                            key={audio.id}
+                            controls
+                            className={audioTagClassNames}
+                            controlsList="nodownload"
+                            preload="auto"
+                        >
+                            <source src={audio.genericUrl} type="audio/mp3" />
```

Comment:
You can't assume `genericUrl` to be an `mp3`, it can be a `wav` or `ogg` or any other format, what about:
```tsx
<audio
    data-test-id="audio-block-audio-tag"
    key={audio.id}
    controls
    className={audioTagClassNames}
    controlsList="nodownload"
    preload="auto"
    src={audio.genericUrl}
/>
```